src/main.rs

```diff
@@ -60,10 +63,49 @@ pub struct UploadResponse {
     url: String,
 }
 
+struct DirEntryModTimePair {
+    dir_entry: DirEntry,
+    mod_time: SystemTime,
+}
+
 async fn index() -> impl Responder {
     HttpResponse::Ok().body("i API ready!")
 }
 
+async fn recent(opt: web::Data<Opt>) -> impl Responder {
+    let mut files = Vec::new();
+
+    let base_dir = match get_base_dir(&opt) {
+        Ok(x) => x.to_string(),
+        Err(_) => panic!("Cant get base dir!"),
+    };
+
+    // Apperently I need to use the result of visit_dirs
+    match visit_dirs(Path::new(&base_dir), &mut files) {
+        Ok(_) => {}
+        Err(_) => panic!("SystemTime before UNIX EPOCH!"),
+    };
+
+    // note the order of the partial_cmp
+    files.sort_by(|a, b| b.mod_time.partial_cmp(&a.mod_time).unwrap());
+
+    let n_of_recent_files = 5;
+    let n_of_recent_files = if n_of_recent_files < files.len() {
+        n_of_recent_files
+    } else {
+        files.len()
+    };
+    let mut latest_n_files: Vec<&DirEntryModTimePair> = Vec::new();
+
+    for file in files.iter().take(n_of_recent_files) {
```

Comment:
I _think_ a more idiomatic rust construction would be to use `collect()` instead of a for-loop with push.

`let latest_n_files: Vec<&DirEntryModTimePair> = files.iter().take(n_of_recent_files).collect();`

Also, I got curious and searched on `take()`, and it seems it takes _up to_ `n_of_recent_files` elements, so I don't think we need the if statement above :tada: (I just learned this! Code review is great! :slightly_smiling_face: )